image_writers_guide/guidelines.adoc

```diff
@@ -9,4 +9,85 @@
 
 toc::[]
 
-== Overview
\ No newline at end of file
+== Overview
+When creating Docker images to run on OpenShift there are a number of best practices to consider in order to ensure consumers of those images have a good experience.  Since images are intended to be immutable and used as is, it is the image author's responsibility to ensure the image contains any artifacts that would typically be needed by a user of the image and that the image can be configured by the user.  Here are some general comments around things to consider.
+
+== General Docker Best Practices
+The following are some guidelines that apply to Docker image creation in general, independent of whether the images are used on OpenShift v3.
+
+=== Reuse images
+Whereever possible you should base your image on an appropriate upstream image via the `FROM` statement.  This will ensure that your image can easily pick up security fixes
+from an upstream image when it is updated, rather than you being responsible for updating your dependencies directly.  
+
+In addition, you should use tags in the `FROM` instruction (for example `rhel:rhel7`) will make it clear to users exactly which version of an image your image is based on.  If you use a tag other than `latest` this ensures that your image is not subjected to breaking changes that might go into the `latest` version of an upstream image.
+
+=== Do not start multiple processes
+It is not a good idea to start multiple services (like database, sshd, …​) inside one container. Containers are lighweight and can be easily linked together for multiprocess orchestration.  This also makes updates less disruptive as each image can be updated less frequently and independently.  Having a single process also makes signal handling flows clearer as you do not need to manage routing signals to spawned processes.
+
+=== Use exec in wrapper scripts
+See http://www.projectatomic.io/docs/docker-image-author-guidance/, subsection _Always `exec` in Wrapper Scripts_
+
+=== Clean temporary files
+Every temporary files you create during the build process should be removed. This includes any files added with the ADD command too.  For example it is strongly recommended you run yum clean after performing yum install operations.  If you craft your `RUN` statement as follows, you an prevent the yum cache from ending up in an image layer:
```

Comment:
I would add "run all of your package install steps in one RUN command, because yum clean all -y won't clean up things added in previous RUNs".  I.e.:

```
RUN yum install stuff && \
    yum clean all -y
```